pushflatpakscript/src/pushflatpakscript/flathub.py

```diff
@@ -138,6 +138,13 @@ def files(tar, path):
     return os.path.join(flatpak_tar_basedir, flatpak_deflated_dir)
 
 
+def check_app_id_matches_flatpak(context, flatpak_path):
+    flatpak_app_ref_dir = flatpak_path + "/refs/heads/app/" + context.config["app_id"]
+
+    if not os.path.exists(flatpak_app_ref_dir):
```

Comment:
Is this right?  It seems as though you could craft a flatpak with both legit and bogus app refs and this check would be happy?
I'd be more defensive here and ensure that is the only app in the repo (which maybe means we need to install `ostree` in the pushflatpakscript image so we can run commands against it).

Reply:
It turns out ostree was already available in the pushflatpakscript Docker image, so I updated this patch to use ostree to verify that the correct app is present in the Flatpak, and that no other apps are present.